api/base/serializers.py

```diff
@@ -225,7 +225,7 @@ def to_internal_value(self, data):
             if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
                 id_field = getattr(self.root.instance, self.source, '_id')
                 if id_field != data:
-                    raise Conflict()
+                    raise Conflict(detail=('The id you specified in the URL "' + id_field + '" exists, but does not match the id you specified in the JSON body "' + data + '".'))
```

Comment:
Unless you checked and I'm wrong about the 404, in which case let me know and disregard all these extra messages.